### PR TITLE
Newapp: Expose multiple ports in generated services

### PR DESCRIPTION
**Pull request by csrwng**
Follow-up to https://github.com/openshift/origin/pull/3201

### CODE REVIEW

**Comment by csrwng** [comment]
@deads2k ptal

**Comment by csrwng** [comment]
[test]

File: pkg/generate/app/pipeline.go [hunk 1]
```diff
@@ -190,6 +190,10 @@ func (s sortablePorts) Swap(i, j int) {
 	s[j] = p
 }
 
+func portKey(port int, protocol kapi.Protocol) string {
+	return fmt.Sprintf("%d/%s", port, protocol)
```
Comment (deads2k) [inline pkg/generate/app/pipeline.go]: Why not have this use the `"port-%d-%s"` format as well?

**Comment by deads2k** [comment]
gofmt

**Comment by deads2k** [comment]
lgtm

File: pkg/generate/app/pipeline.go (hunk 1 above)
Comment (csrwng) [inline pkg/generate/app/pipeline.go]: it really doesn't matter, it's just to make sure that a port is unique... in the Dockerfile, there's specified as [num]/[protocol] and that's what I followed here.

**Comment by csrwng** [comment]
gofmt fixed

**Comment by csrwng** [comment]
[test]

**Comment by csrwng** [comment]
@deads2k, @smarterclayton is this ok to merge?

**Comment by smarterclayton** [comment]
[merge]

**Pull request merged**